### ISSUE TITLE
[BUG] Screenshot captures desktop instead of fullscreen window.

### ISSUE THREAD

**Issue by Draconicrose** [linuxmint/cinnamon#11127]
```
 * Cinnamon 5.4.11
 * Distribution - Mint 21
 * Nvidia GeForce GTX 1660, driver version 515
 * 64 bit
 ```

### Issue

- When taking a screenshot of an area of a fullscreen (or borderless fullscreen) window using the hotkey (shift+printscreen), the screenshot taken shows my desktop instead of the window. 
- ~~Opening the screenshot tool from the cinnamon menu and using the GUI to take an area screenshot captures the window properly.~~ This appears to have been a fluke.

### Steps to reproduce

1. Run a fullscreen or borderless fullscreen window. I usually use video games.
2. Press shift+printscreen to take the screenshot of an area.
3. Select the area to screenshot.
4. Look at the preview
5. Optional: paste in GIMP to confirm that it only captured the desktop.

### Expected behaviour

- Using the hotkey grabs what is seen on the screen instead of just the desktop.

### Other information
* [xsession-errors.txt](https://github.com/linuxmint/cinnamon/files/9465456/xsession-errors.txt)
* My keyboard is a 65% keyboard so I have to actually use shift+Fn+' to get the equivalent of shift+printscreen.
* Doing a clean reinstall of gnome-screenshot fixed the problem for a single screenshot, then it returned.

### Configuration info, if applicable
N/A

**Comment by ZhaoZuohong** [comment]
Linux Mint 21 Cinnamon 5.4.12, same issue here.
When taking screenshot of full-screen wine applications, gnome-screenshot can only get desktop instead of the window.
Taking screenshot under the same condition with other tools (for example, flameshot) works.

**Comment by ZhaoZuohong** [comment]
Here is a workaround for this bug.

Create a shell script, eg. `~/bin/screenshot.sh` with following content:

```bash
#!/bin/bash

filename=~/Pictures/screenshot-$(date +%Y%m%d-%H%M%S).png
import -window root $filename
cat $filename | xclip -selection clipboard -t image/png
notify-send -i gnome-screenshot "Screenshot" "Screenshot saved as $filename"
```

You can give it executable permission with `chmod +x screenshot.sh` and test it in a terminal. If it works file, the next step is to bind it to the PrtScn key.

Goto system settings → keyboard → shortcuts, delete default binding for taking full screen shotcut, and add a new one:

![image](https://user-images.githubusercontent.com/34163622/201916722-446435f6-f416-4920-bfa7-de3130577ca6.png)

![image](https://user-images.githubusercontent.com/34163622/201917027-99cef3e6-27d5-4cb7-b1ed-5b98ec48e276.png)

The *command* field should be something like `bash <path_to_your_script.sh>`.

**Comment by Draconicrose** [comment]
Still present as of Cinnamon 5.6.5. Also found that it "works" when using alt + printscreen. The game screen is captured but Cinnamon crashes. Here's the log: https://termbin.com/rhwg

**Comment by Draconicrose** [comment]
Appears fixed in Cinnamon 5.6.7!

**Comment by claudiux** [comment]
This bug seems indeed fixed in Cinnamon 5.6.7.

**Comment by Draconicrose** [comment]
> 

It's not, it is just way less frequent! Had it happen to me twice just yesterday. Plus, using the alt + printscrn shortcut still makes Cinnamon crash. I'm assuming these are related, but it could be something else.
